### PR TITLE
Enable JSON encode pretty print

### LINKED ISSUE #1139: Ability for pretty JSON in NormalizerFormatter
Now it is very difficult to make pretty formatting of the JSON. The toJson method is private and it not support JSON_PRETTY_PRINT option. 

How to optimally make the data in the extra processed using the JSON_PRETTY_PRINT option?

### PR DESCRIPTION
**Pull request by gmponos**
Closes #1139

### CODE REVIEW

File: src/Monolog/Formatter/NormalizerFormatter.php [hunk 1]
```diff
@@ -89,6 +91,14 @@ public function setMaxNormalizeItemCount(int $maxNormalizeItemCount): self
         return $this;
     }
 
+    /**
+     * Enables `json_encode` pretty print.
+     */
+    public function enablePrettyPrint(): void
```
[the remark this answers is not in the document]
Comment (gmponos) [reply src/Monolog/Formatter/NormalizerFormatter.php]: hmmm.. I did the change.. although I am not sure I like how it is now... let me know what you think.

[the remark this answers is not in the document]
Comment (Seldaek) [reply src/Monolog/Formatter/NormalizerFormatter.php]: What don't you like? :) The only improvement I could see is to set it like:

```
if ($enable) {
    $this->jsonEncodeOptions |= JSON_PRETTY_PRINT;
} else {
    $this->jsonEncodeOptions ^= JSON_PRETTY_PRINT;
}
```

So it adds/remove the flag and you don't need the default const. But it's kind of a nitpick.

[the remark this answers is not in the document]
Comment (gmponos) [reply src/Monolog/Formatter/NormalizerFormatter.php]: done....

> What don't you like?

nvm... it's fine.. I believe we reached to the best solution :)

**Comment by Seldaek** [comment]
Ok thanks!

**Pull request merged**